test/compose.c

```diff
@@ -588,6 +588,16 @@ main(int argc, char *argv[])
     ctx = test_get_context(CONTEXT_NO_FLAG);
     assert(ctx);
 
+    /*
+     * Ensure the environment variables that configure Compose file paths are
+     * unset before the tests.
+     * They are set explicitely in the corresponding tests when necessary.
+     */
+    unsetenv("XCOMPOSEFILE");
+    unsetenv("XDG_CONFIG_HOME");
+    unsetenv("HOME");
+    unsetenv("XLOCALEDIR");
```

Comment:
Stared at this and then thought it's a whack-a-mole. So I tested this hunk locally and behold, it works!

```c
    const char *srcdir = getenv("top_srcdir");
    clearenv();
    setenv("top_srcdir", srcdir, 1);
```
apparently the only env var we need here is `top_srcdir`.

Reply:
Done

Reply:
`clearenv` is not available on all platforms 😒

Fixed